bloom-housing/ui-seeds
# refactor: upgrade storybook

## Description

Upgrades Storybook to be able to support some of the documentation requests from SF.

This PR 1) Upgrades Storybook + documentation, and (2) introduces new shared types that are used across components instead of string unions (none of these are breaking changes, but having importable types has come up a few times as well).

Each code change unrelated to this has a comment on it. This includes two separate changes that are breaking, which is to rename the `highlight` variant of Button to `warn`, and the `in-process` variant of Tag to `warn`, to match the tokens and designs. (UPDATE: Now also includes renaming instances of accent to highlight!)

## How Can This Be Tested/Reviewed?

The huge diff is mostly in the lock file.

* Changes in mdx and stories files are for managing the upgrade, improving organization, and enabling code samples
* Changes in Storybook configuration files are for managing the upgrade
* Changes in .tsx files are for improving typing

Compare the [old Storybook](https://storybook-ui-seeds.netlify.app/?path=/story/tokens-introduction--page) to the [deploy preview ](https://deploy-preview-93--storybook-ui-seeds.netlify.app/?path=/docs/tokens-introduction--docs) here to see changes and ensure things are looking good! The Button and Tag stories will be the only ones w the naming adjustment to `warn`.

There are now
* Code samples for every story
* Links to Zeroheight

## Checklist:
